### ISSUE TITLE
Crash when closing and restoring timeline dock

### ISSUE THREAD

**Issue by jmcgvr** [mltframework/shotcut#151]
To reproduce
- open any existing mlt file that has any kind of track in the timeline. Just opening shotcut with an untitled project then adding a track doesn't seem to ever cause this.
- close the timeline dock, reopen it, crash
- this isn't consistent (but very frequent for me). Shotcut seems to be able to "settle" into a state where no number of closing and reopening the timeline dock causes a crash, including loading other mlt files.
- If it doesn't crash, the audio waveforms are not redrawn in the timeline
```
Problem signature:
  Problem Event Name:	APPCRASH
  Application Name:	shotcut.exe
  Application Version:	0.0.0.0
  Application Timestamp:	565fbaaa
  Fault Module Name:	Qt5Quick.dll
  Fault Module Version:	5.5.0.0
  Fault Module Timestamp:	559bf2b5
  Exception Code:	c0000005
  Exception Offset:	0000000000059e25
  OS Version:	6.1.7601.2.1.0.256.1
  Locale ID:	1033
  Additional Information 1:	1ef3
  Additional Information 2:	1ef3fde1435c37b31f094ed73efc4e41
  Additional Information 3:	74a4
  Additional Information 4:	74a4a12429940639514f8911e1d99c14
```

**Comment by ddennedy** [comment]
I reproduced it on Windows and Linux.

**Comment by ddennedy** [comment]
Crashes in QTBUG47714WorkaroundRenderListener::afterSync() on `delete oldTexture`.

**Comment by metellius** [comment]
Currently working on this.

**Comment by ddennedy** [comment]
Fixed in 1dd6491.
The other linked commit 27f58a3 does not apply (commit message changed just after noticing the mistake.)
